dozer-ingestion/src/connectors/object_store/table_reader.rs

```diff
@@ -147,27 +147,30 @@ impl<T: DozerObjectStore> Reader<T> for TableReader<T> {
             let sender = tx.clone();
             let t = table.clone();
 
-            thread::spawn(move || {
-                let result = rt.block_on(Self::read(
+            rt.spawn(async move {
+                let result = Self::read(
                     id as u32,
                     ctx,
                     table_path,
                     listing_options,
                     &t,
                     sender.clone(),
-                ));
-
+                )
+                .await;
                 if let Err(e) = result {
                     sender.send(Err(e)).unwrap();
+                    return Err(ObjectStoreConnectorError::DataReadError);
```

Comment:
What's the purpose of returning error here?

Reply:
Right, there is no purpose. Removed it in last commit.

Reply:
Also remove `DataReadError`?